pkg/app/AppService.go

```diff
@@ -2186,7 +2259,20 @@ func (impl *AppServiceImpl) GetConfigMapAndSecretJson(appId int, envId int, pipe
 	return merged, nil
 }
 
-func (impl *AppServiceImpl) getConfigMapAndSecretJsonV2(appId int, envId int, pipelineId int, chartVersion string, deploymentWithConfig bean.DeploymentConfigurationType, wfrIdForDeploymentWithSpecificTrigger int) ([]byte, error) {
+type CMCSJsonDTO struct {
+	AppId                                 int                              `json:"appId"`
+	EnvId                                 int                              `json:"envId"`
+	PipeLineId                            int                              `json:"pipeLineId"`
+	ChartVersion                          string                           `json:"chartVersion"`
+	DeploymentWithConfig                  bean.DeploymentConfigurationType `json:"deploymentWithConfig"`
+	wfrIdForDeploymentWithSpecificTrigger int
+	ImageTag                              string `json:"imageTag"`
+	AppName                               string `json:"appName"`
+	envOverride                           *chartConfig.EnvConfigOverride
+	Image                                 string `json:"image"`
```

Comment:
create a scope object and pass, since you're adding just one extra argument, let's not create this request struct for now